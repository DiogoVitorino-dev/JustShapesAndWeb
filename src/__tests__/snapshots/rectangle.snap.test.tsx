import { render, renderHook } from "@testing-library/react-native";
import { useSharedValue } from "react-native-reanimated";

import Rectangle, {
  RectangleAngle,
  RectanglePosition,
  RectangleSize,
} from "@/models/geometric/rectangle";

describe("Rectangle model - Snapshot test", () => {
  it("Should renders correctly", () => {
    let pos: RectanglePosition;
    let angle: RectangleAngle;
    let size: RectangleSize;

    renderHook(() => {
      pos = useSharedValue({ x: 25, y: 50 });
      angle = useSharedValue(90);
      size = useSharedValue({ width: 100, height: 50 });
    });

    //@ts-expect-error
    const tree = render(<Rectangle angle={angle} position={pos} size={size} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
