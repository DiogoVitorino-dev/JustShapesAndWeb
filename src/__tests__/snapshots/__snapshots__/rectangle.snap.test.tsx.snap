// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rectangle model - snapshot test Should renders correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "height": 50,
        "left": 25,
        "top": 50,
        "transform": [
          {
            "rotate": "90deg",
          },
        ],
        "width": 100,
      },
    }
  }
  collapsable={false}
  style={
    {
      "height": 50,
      "left": 25,
      "position": "absolute",
      "top": 50,
      "transform": [
        {
          "rotate": "90deg",
        },
      ],
      "width": 100,
    }
  }
  testID="rectangleModel"
/>
`;
